fedimint-testing/src/btc/mock.rs

```diff
@@ -205,26 +206,33 @@ impl BitcoinTest for FakeBitcoinTest {
     }
 
     async fn get_mempool_tx_fee(&self, txid: &Txid) -> Amount {
-        let pending = self.pending.lock().unwrap();
-        let addresses = self.addresses.lock().unwrap();
-
-        let mut fee = Amount::ZERO;
-        let tx = pending
-            .iter()
-            .find(|tx| tx.txid() == *txid)
-            .expect("tx was broadcast");
-
-        for input in tx.input.iter() {
-            fee += *addresses
-                .get(&input.previous_output.txid)
-                .expect("tx has sats");
-        }
+        loop {
+            let pending = self.pending.lock().unwrap().clone();
```

Comment:
Is there any need to hold these locks at the same time? Leaves possibility for deadlock open and AFAICT, it could be avoided.